File: 2833-furthest-point-from-origin/2833-furthest-point-from-origin.go
func furthestDistanceFromOrigin(moves string) int {
    r, l, x := 0, 0, 0
    for _, move := range moves {
        if move == 'R' {
            r++
        } else if move == 'L' {
            l++
        } else {
            x++
        }
    }

    return max(r, l) - min(r, l) + x
}

// counting, string
// time: O(n)
// space: O(1)

File: 2833-furthest-point-from-origin/2833-furthest-point-from-origin.py
class Solution:
    def furthestDistanceFromOrigin(self, moves: str) -> int:
        r, l, x = 0, 0, 0
        for m in moves:
            if m == 'R':
                r += 1
            elif m == 'L':
                l += 1
            else:
                x += 1
        
        return max(r, l) - min(r, l) + x

# counting, string
# time: O(n)
# space: O(1)